cmapi/cmapi_server/controllers/s3dataload.py

```diff
@@ -288,24 +288,32 @@ def prepare_google_storage(
         cpimport_error = ''
         cpimport_output = ''
 
-        alive = 3
-        while alive > 0:
+        while True:
             events = selector.select()
             for key, mask in events:
                 name = key.data
                 line = key.fileobj.readline().rstrip()
-                if not line:
-                    # EOF
-                    alive -= 1
-                    selector.unregister(key.fileobj)
-                    continue
-                if name == 'downloader_error':
+                if name == 'downloader_error' and line:
                     downloader_error += line + '\n'
-                if name == 'cpimport_error':
+                if name == 'cpimport_error' and line:
                     cpimport_error += line + '\n'
-                if name == 'cpimport_output':
+                if name == 'cpimport_output' and line:
                     cpimport_output += line + '\n'
 
+            if downloader_error:
```

Comment:
If we have `downloader_error` we'll never know about `cpimport_error`.

Reply:
That's sounds good, downloader error is premature for cpimport